x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/SearchableSnapshotShardStats.java

```diff
@@ -395,4 +396,55 @@ public int hashCode() {
             return Objects.hash(count, total, min, max);
         }
     }
+
+    public static class TimedCounter extends Counter {
+
+        private final long totalNanoseconds;
+
+        public TimedCounter(long count, long total, long min, long max, long totalNanoseconds) {
+            super(count, total, min, max);
+            this.totalNanoseconds = totalNanoseconds;
+        }
+
+        TimedCounter(StreamInput in) throws IOException {
+            super(in);
+            totalNanoseconds = in.readZLong();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeZLong(totalNanoseconds);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
Maybe define an empty `innerToXContent(XContentBuilder, Params)` to the `Counter` class and overrides it here to only add the `time` field? That's what we do for other XContent implementations.

Reply:
Nice, thanks for the tip. Done in 5b26f66.